components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/impl/WSDL20ProcessorImpl.java

```diff
@@ -0,0 +1,333 @@
+                    log.debug("Successfully updated endpoints of WSDL: " + entry.getKey());
+                }
+                try (FileOutputStream wsdlFileOutputStream = new FileOutputStream(new File(entry.getKey()))) {
+                    WSDLWriter writer = getWsdlFactoryInstance().newWSDLWriter();
+                    writer.writeWSDL(description.toElement(), wsdlFileOutputStream);
+                } catch (IOException | WSDLException e) {
+                    throw new APIMgtWSDLException(
+                            "Failed to create WSDL archive for API:" + api.getName() + ":" + api.getVersion()
+                                    + " for label " + label.getName(),
+                            ExceptionCodes.ERROR_WHILE_CREATING_WSDL_ARCHIVE);
+                }
+            }
+        }
+        return wsdlArchiveExtractedPath;
+    }
+
+    @Override
+    public boolean canProcess() {
+        return canProcess;
+    }
+
+    @Override
+    public WSDLInfo getWsdlInfo() throws APIMgtWSDLException {
+        Map<String, String> endpointsMap = getEndpoints();
+        WSDLInfo wsdlInfo = new WSDLInfo();
+        wsdlInfo.setEndpoints(endpointsMap);
+        wsdlInfo.setVersion(APIMgtConstants.WSDLConstants.WSDL_VERSION_20);
+        return wsdlInfo;
+    }
+
+    /**
+     * Updates the endpoints of the {@code description} based on the provided {@code endpointURLs} and {@code api}.
+     *
+     * @param endpointURLs Endpoint URIs
+     * @param api Provided API object
+     * @param description WSDL 2.0 description
+     * @throws APIMgtWSDLException If an error occurred while updating endpoints
+     */
+    private void updateEndpoints(List<String> endpointURLs, API api, Description description)
+            throws APIMgtWSDLException {
+        String context = api.getContext().startsWith("/") ? api.getContext() : "/" + api.getContext();
+        String selectedUrl;
+        try {
+            selectedUrl = APIMWSDLUtils.getSelectedEndpoint(endpointURLs) + context;
+            if (log.isDebugEnabled()) {
+                log.debug("Selected URL for updating endpoints of WSDL: " + selectedUrl);
+            }
+        } catch (MalformedURLException e) {
+            throw new APIMgtWSDLException("Error while selecting endpoints for WSDL", e,
+                    ExceptionCodes.INTERNAL_WSDL_EXCEPTION);
+        }
+        if (!StringUtils.isBlank(selectedUrl)) {
+            updateEndpoints(selectedUrl, description);
+        }
+    }
+
+    /**
+     * Clear the actual service endpoints in {@code description} and use {@code selectedEndpoint} instead of 
+     * actual endpoints.
+     *
+     * @param selectedEndpoint Endpoint which will replace the WSDL endpoints
+     * @param description WSDL 2.0 description
+     * @throws APIMgtWSDLException If an error occurred while updating endpoints
+     */
+    private void updateEndpoints(String selectedEndpoint, Description description) throws APIMgtWSDLException {
```

Comment:
Nope, this needs to be fixed. Thanks for the pointer. Needs to catch the proper exception and throw an APIMgtWSDLException.